simulation/vision.py

```diff
@@ -112,25 +116,39 @@ def line(x0, y0, x1, y1):
     # same but without numpy:       0.4 ms / call
     # proper version:               0.6 ms / call
     def _reveal_visible(self, x0: int, y0: int, radius: float, view_angle: float, heading: float, in_tower: bool):
-        offset = int(math.ceil(radius)) + 1
+        vision_modifier = self._map.get_vision_modifier(x0, y0)
+        offset = int(math.ceil(18*vision_modifier))  # + 1
```

Comment:
What's the 18?

Reply:
sorry, i should have made that clearer in the code. agents can see towers from 18 m away (i assume also through walls since they should be higher). they can also see structures from 10 m away if theer's a line of vision, so basically they can see a wall they are walking towards before they can see the ground in front of that wall